kingpin/actors/aws/ecs.py

```diff
@@ -181,19 +174,113 @@
 class ECSBaseActor(base.AWSBaseActor):
     """Base class for ECS actors."""
 
-    def _handle_failures(self, failures):
+    FAILURE_MISSING = 'MISSING'
+
+    def _handle_failures(self, failures, *ignorable):
         """Logs ECS failures. Raises a recoverable exception if there are any.
 
         Args:
             failures: list of failures from ECS api.
+            ignorable: list of failures to ignore.
         """
         if not failures:
             return
 
-        # TODO: When can we recover? What are the failures? How do we recover?
+        failure_reasons = []
+
         for failure in failures:
+            reason = failure['reason']
+            if reason in ignorable:
+                continue
+            failure_reasons.append(reason)
             self.log.error(failure)
-        raise exceptions.RecoverableActorFailure()
+        if failure_reasons:
+            raise exceptions.RecoverableActorFailure(failure_reasons)
+
+    @gen.coroutine
+    @dry('Would register task definition with family {0[family]}')
+    def _register_task(self, task_definition):
+        """Registers a task on ECS.
+
+        Args:
+            task_definition: dict of ECS task definition parameters.
+
+        Returns:
+            tuple: ('family', 'revision', 'task_definition_name').
+        """
+
+        family = task_definition['family']
+
+        self.log.info('Registering task definition with family {}'.format(
+            family))
+        try:
+            response = yield self.thread(
+                self.ecs_conn.register_task_definition, **task_definition)
+        except Boto3Error as e:
+            raise exceptions.UnrecoverableActorFailure(e.message)
+
+        # Parse data from the server's response.
+        task_definition = response['taskDefinition']
+        family = task_definition['family']
+        revision = task_definition['revision']
+
+        task_definition_name = '{}:{}'.format(family, revision)
+        self.log.info('Task definition {} registered'.format(
+            task_definition_name))
+        raise gen.Return((family, revision, task_definition_name))
+
+    @staticmethod
```

Comment:
Sweet! Well, this method is never used outside of the class at all and is never meant to be. This is true for all the (5) staticmethods defined in these classes. So, the advantage of them being staticmethods is only that they explicitly state that they're purely functional, i.e. they have no side effects.

Originally I had them outside the class entirely but changed that recently, as this makes more sense.